common/epoll.edl

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+/*
+**==============================================================================
+**
+** epoll.edl:
+**
+**     This file declares OCALLs and ECALLs needed by the enclave to implement
```

Comment:
Note that none of the new EDL files actually define ECALLS, and the one ECALL defined in syscalls.edl was to work around the problem that edger8r required at least one ECALL in the EDL file.

This also suggests that despite the refactoring, the only way to use the syscall EDLs is through syscall.edl only. Is that correct, and if so, is that intended?

Reply:
Ah, I didn't realize an ecall is required.

The intent is that these can be important into the EDL file for an enclave application, so they aren't really intended to stand alone. The idea is to make it easy to say "I want to use sockets, all I have to do is `from "socket.edl" import *`". So in that case I think it's fine to have no ecalls.

On that note though, this seems like a bug in edger8r that we should fix. I don't think there's a reason to require an ecall in an enclave application.

Reply:
I agree that this should be fixed in edger8r (maybe it has? @anakrish?). You should just strike the mention of ECALLs in the EDL headers that don't contain them. 

I noticed it because of the presence of the ECALL stub in syscalls.edl and the explanation for it. There needs to be follow-up work to remove that at least, and it would be useful to test that a sample can build against sockets.edl by itself without syscalls.edl before we start promoting that mode of usage.